ddtrace/contrib/starlette/patch.py

```diff
@@ -68,7 +71,16 @@ def unpatch():
 def traced_handler(wrapped, instance, args, kwargs):
     # Since handle can be called multiple times for one request, we take the path of each instance
     # Then combine them at the end to get the correct resource names
-    scope = get_argument_value(args, kwargs, 0, "scope")
+    scope = get_argument_value(args, kwargs, 0, "scope")  # type: Optional[Dict[str, Any]]
+    if not scope:
+        return wrapped(*args, **kwargs)
+
+    # Our ASGI TraceMiddleware has not been called, skip since
+    # we won't have a request span to attach this information onto
+    # DEV: This can happen if patching happens after the app has been created
```

Comment:
is it worth checking each time?
there are so many things broken if you don't patch correctly anyhow 😢

Reply:
The alternative being unpatch this function so we don't keep trying?

You could have multiple Starlette applications (sub-mounts) so it could be possible one application succeeds and another has it.